typing/types.ml

```diff
@@ -575,12 +581,46 @@ let repr t =
      repr_link1 t t'
  | _ -> t
 
+(* scope_field and marks *)
+
+let scope_mask = (1 lsl 27) - 1
+let marks_mask = (-1) lxor scope_mask
+type type_mark =
+  | Mark of {mark: int; mutable marked: type_expr list}
+  | Set of {mutable visited: TransientTypeSet.t}
```

Comment:
Nit: the `visited` field is used in a purely imperative way and could be a `Hashtbl` instead.

Reply:
Indeed. The idea is that this is only a fallback, so performance doesn't really matter.
However our measurement show that using a hash table is more efficient (when there are no marks), so it may be worth switching to reflect that.